### PR TITLE
checking the annotations subtitles formats to be the valid values defined in the properties file

### PR DESCRIPTION
**Pull request by SrdjanStevanetic**
@gsergiu Please have a look. I have noticed that there are some tests that did not pass like the testDereferenceMany(). I did not dive into details. Please let me know if that is necessary to improve at this point.

### CODE REVIEW

File: annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/BaseAnnotationValidator.java [hunk 1]
```diff
@@ -486,11 +487,18 @@ protected void validateSubtitleOrCaption(Annotation webAnnotation)
...
 	Body body = webAnnotation.getBody();
 	validateFullTextResource(body);
-	// check mandatory field edmRights
+	// check mandatory field body.format (please note that this field is saved in the "contentType" field of the given Resource)
 	if (StringUtils.isBlank(body.getContentType())) {
 	    throw new PropertyValidationException(I18nConstants.MESSAGE_MISSING_MANDATORY_FIELD,
 		    I18nConstants.MESSAGE_MISSING_MANDATORY_FIELD, new String[] { "subtitle.body.format" });
 	}
+	// check if the body.format field has a valid value
+	String[] validSubtitlesFormats = getConfiguration().getAnnotationSubtitlesFormats().split(",");
```
Comment (gsergiu) [inline annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/BaseAnnotationValidator.java:496]: this processing of the formats string should happen in Configuration class and the class should return a Set<String>, that that the verification of the format is straight forward and performant

**Review by gsergiu — CHANGES_REQUESTED** [review]
the processing of the formats needs to be done only once when loading configurations

File: annotation-utils/src/main/java/eu/europeana/annotation/config/AnnotationConfigurationImpl.java [hunk 2]
```diff
@@ -47,6 +47,12 @@ public String getAnnotationBaseUrl() {
 	String key = ANNOTATION_ENVIRONMENT + "." + getEnvironment() + "." + SUFFIX_BASEURL;
 	return getAnnotationProperties().getProperty(key);
     }
+    
+    @Override
+    public String getAnnotationSubtitlesFormats() {
```
[the remark this answers is not in the document]
Comment (SrdjanStevanetic) [reply annotation-utils/src/main/java/eu/europeana/annotation/config/AnnotationConfigurationImpl.java]: Improved.

File: annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/BaseAnnotationValidator.java (hunk 1 above)
Comment (SrdjanStevanetic) [reply annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/BaseAnnotationValidator.java]: Improved.

**Pull request merged**